src/components/Trade/hooks/useReceiveAddress.tsx

```diff
@@ -0,0 +1,82 @@
+import type { Asset } from '@shapeshiftoss/asset-service'
+import { fromAccountId } from '@shapeshiftoss/caip'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useFormContext, useWatch } from 'react-hook-form'
+import { getReceiveAddress } from 'components/Trade/hooks/useSwapper/utils'
+import type { TS } from 'components/Trade/types'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import {
+  selectPortfolioAccountIdsByAssetId,
+  selectPortfolioAccountMetadataByAccountId,
+} from 'state/slices/portfolioSlice/selectors'
+import { isUtxoAccountId } from 'state/slices/portfolioSlice/utils'
+import { useAppSelector } from 'state/store'
+
+export const useReceiveAddress = () => {
```

Comment:
Existing logic - extracted into a dedicated custom hook to separate important logic, and prevent circular imports.